pkg/api/customization/cluster/validator.go

```diff
@@ -270,3 +275,72 @@ func (v *Validator) accessTemplate(request *types.APIContext, spec *mgmtclient.C
 
 	return nil
 }
+
+// validateGenericEngineConfig allows for additional validation of clusters that depend on Kontainer Engine or Rancher Machine driver
+func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec *v3.ClusterSpec) error {
+
+	if request.Method == http.MethodPost {
+		return nil
+	}
+
+	clusterName := request.ID
+	prevCluster, err := v.ClusterLister.Get("", clusterName)
+	if err != nil {
+		return httperror.WrapAPIError(err, httperror.InvalidBodyContent, err.Error())
+	}
+	if spec.AmazonElasticContainerServiceConfig != nil {
+		err := validateEKS(*prevCluster.Spec.GenericEngineConfig, *spec.AmazonElasticContainerServiceConfig)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+
+}
+
+func validateEKS(prevCluster, newCluster map[string]interface{}) error {
+	// check config is for EKS clusters
+	if driver, ok := prevCluster["driverName"]; ok {
+		if driver != service.AmazonElasticContainerServiceDriverName {
+			return nil
+		}
+	}
+
+	// don't allow for updating subnets
+	if prev, ok := prevCluster["subnets"]; ok {
+		if new, ok := newCluster["subnets"]; ok {
+			prevSlice, ok := prev.([]interface{})
+			if !ok {
+				return httperror.NewAPIError(httperror.InvalidBodyContent, "unable to convert subnets to interace slice")
+			}
+			newSlice, ok := new.([]interface{})
+			if !ok {
+				return httperror.NewAPIError(httperror.InvalidBodyContent, "unable to convert subnets to interace slice")
```

Comment:
Same comment as above. Has it been tested that we can shove through something that would actually hit this error? The test cases below don't test that.

Reply:
I think it is unlikely this will occur but I would prefer to catch it instead of letting it panic. Returning no error might allow a potentially badly form request to get further through the api

Reply:
Dropped